File: test_extension.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
  const extensionPath = path.join(__dirname, 'dist');

  const browser = await chromium.launchPersistentContext('', {
    headless: false,
    args: [
      `--disable-extensions-except=${extensionPath}`,
      `--load-extension=${extensionPath}`
    ]
  });

  const page = await browser.newPage();
  await page.goto('https://www.youtube.com/watch?v=dQw4w9WgXcQ');

  // Wait for video to load
  await page.waitForTimeout(3000);

  // Take screenshot before opening extension
  await page.screenshot({ path: 'before_extension.png' });
  console.log('Navigated to YouTube, waiting for you to click extension icon...');

  // Wait for extension to be activated
  await page.waitForTimeout(30000);

  // Take screenshot after extension is open
  await page.screenshot({ path: 'after_extension.png' });

  console.log('Screenshots saved. Check before_extension.png and after_extension.png');

  await browser.close();
})();

File: test_extension.py
from playwright.sync_api import sync_playwright
import time
import os

extension_path = os.path.join(os.getcwd(), 'dist')
print(f"Loading extension from: {extension_path}")

with sync_playwright() as p:
    # Launch browser with extension
    context = p.chromium.launch_persistent_context(
        "",
        headless=False,
        args=[
            f'--disable-extensions-except={extension_path}',
            f'--load-extension={extension_path}'
        ]
    )

    page = context.new_page()
    page.goto('https://www.youtube.com/watch?v=dQw4w9WgXcQ')

    print("Opened YouTube. Click the extension icon to test it.")
    print("Waiting 30 seconds for you to interact...")

    time.sleep(30)

    # Take screenshot
    page.screenshot(path='extension_test.png')
    print("Screenshot saved as extension_test.png")

    context.close()
